src/content/1.7/modules/concepts/services/_index.md

```diff
@@ -154,6 +161,82 @@ If you need more details about dependency injection and how services work in the
 their documentation about the [Service Container](https://symfony.com/doc/3.4/service_container.html).
 {{% /notice %}}
 
+### Override an existing Symfony service
+
+The container definition can be modified by a module, which enables you to override an existing Symfony service being used in PrestaShop.
+
+This is a mechanism similar to PrestaShop standard overrides, but the main benefit is that the php code stays unmodified. This prevents issues linked to code definition or autoloading failures.
+
+As you can read it from the [Symfony documentation](https://symfony.com/doc/current/service_container/service_decoration.html), there are 2 ways to modify an existing service:
+
+#### Override the service
+
+When you choose to override a service, this means that you _replace the service by another one_. The previous service is not usable anymore. Every other part of the code where this service is used will use the new version.
+
+To do it: you declare your new service using the old service name. So if you want to override the service `prestashop.core.b2b.b2b_feature` with your own implementation, you write in `config/services.yml` :
+
+```yml
+  prestashop.core.b2b.b2b_feature:
+    class: 'YourCompany\YourModule\YourService'
+```
+
+That's done. The service registered under the name `prestashop.core.b2b.b2b_feature` is now your service. The previous `prestashop.core.b2b.b2b_feature` is gone.
+
+#### Decorate the service
```

Comment:
I have a question, about decoration Is it possible for multiple modules to decorate the same service
Will the decoration stack on top of each other then?
If so then we should encourage using this method as it would allow multiple modules to coexist.

The problem we have today with class override will be the same with service override, so anyway we should add a warning notice about this drawback no?

Reply:
Indeed, whether it's PrestaShop class override or Symfony service override, the mechanism is powerful and consequently can break a shop.

Yes, I think you can decorate multiple times a service 🤔 but I dont know which one will get aliased to `inner` 😅

Reply:
In fact, service decoration or service override are both better than PrestaShop class override because **the php class remains untouched**.

So if we have 2 services which use the same class:
```
    my_service_one:
        class: 'MyClass'

    my_service_two:
        class: 'MyClass'
```
Then overriding or decorating `my_service_one` will not break or change the behavior of `my_service_two`. PrestaShop standard override mechanism will do it, on the contrary.